service/Microsoft.Azure.Devices/ServiceClient.cs

```diff
@@ -44,6 +45,26 @@ public static ServiceClient CreateFromConnectionString(string connectionString)
             return CreateFromConnectionString(connectionString, TransportType.Amqp);
         }
 
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases unmanaged and - optionally - managed resources.
+        /// </summary>
+        /// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if ( disposing )
+            {
+                // Call CloseAsync() synchronously avoiding AggregateException.
+                this.CloseAsync().GetAwaiter().GetResult();
+            }
+        }
```

Comment:
I think these base implementations are required because they are `public` (they can have another implementation than genuine Service SDK) and fallback implementation should exist. What do you think about this?